File: libs/dotnet/Exchange.Trading.Domain/Entities/Trade.cs
using Exchange.Trading.Domain.ValueObjects;

namespace Exchange.Trading.Domain.Entities;

public sealed class Trade
{
    public Trade(
        Guid tradeId,
        Guid buyOrderId,
        Guid sellOrderId,
        Guid? instrumentId,
        Guid? buyTradingAccountId,
        Guid? sellTradingAccountId,
        Symbol symbol,
        Price price,
        Quantity quantity,
        DateTimeOffset executedAt)
    {
        TradeId = tradeId;
        BuyOrderId = buyOrderId;
        SellOrderId = sellOrderId;
        InstrumentId = instrumentId;
        BuyTradingAccountId = buyTradingAccountId;
        SellTradingAccountId = sellTradingAccountId;
        Symbol = symbol;
        Price = price;
        Quantity = quantity;
        ExecutedAt = executedAt;
    }

    public Guid TradeId { get; }

    public Guid BuyOrderId { get; }

    public Guid SellOrderId { get; }

    public Guid? InstrumentId { get; }

    public Guid? BuyTradingAccountId { get; }

    public Guid? SellTradingAccountId { get; }

    public Symbol Symbol { get; }

    public Price Price { get; }

    public Quantity Quantity { get; }

    public DateTimeOffset ExecutedAt { get; }
}

File: apps/matching-engine/src/domain/trade.rs
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Trade {
    pub trade_id: String,
    pub instrument_id: Option<String>,
    pub symbol: String,
    pub buy_order_id: String,
    pub sell_order_id: String,
    pub buy_account_id: String,
    pub sell_account_id: String,
    pub buy_trading_account_id: Option<String>,
    pub sell_trading_account_id: Option<String>,
    pub price: u64,
    pub quantity: u64,
}
